vms/proposervm/batched_vm.go

```diff
@@ -155,6 +154,44 @@ func (vm *VM) BatchedParseBlock(ctx context.Context, blks [][]byte) ([]snowman.B
 	return blocks, nil
 }
 
+type parseResults struct {
+	block statelessblock.Block
+	err   error
+}
+
+func (vm *VM) parseBlocks(blks [][]byte) []parseResults {
+	if len(blks) == 0 {
+		return nil
+	}
+
+	results := make([]parseResults, len(blks))
+
+	// If we can't parse the first block, no point in parsing anything after
```

Comment:
The idea is that you get a bunch of blocks upon a response for a message, and they should all be within proximity. 

So if the first cannot be parsed, it's probable the others most likely also can't, and vice versa. 

It stems from the fact that blocks are produced in a certain way starting from a period in time.

Reply:
I still would not special case the first block and venture to parse them all in parallel, instead of serializing parsing to first block one.
This unless we have proof that block parsing fails often and fails mostily because of the first block

Reply:
I think unless we've run a mainnet benchmark with results showing the opposite, I'd probably prefer to just parallelize all of the parsing rather than special casing the first block.

Reply:
Got it, thanks. Changed accordingly!